runner/cf-driver-go/cmd/drive/cleanup.go

```diff
@@ -0,0 +1,37 @@
+package drive
+
+import (
+	"log"
+
+	"github.com/spf13/cobra"
+)
+
+var cleanupCmd = &cobra.Command{
+	Use:   "cleanup",
+	Short: "Cleans up resources, containers, etc. at job completion",
+	Long: `The Cleanup stage is executed by "cleanup_exec".
+
+This final stage is executed even if one of the previous stages failed.
+The main goal for this stage is to clean up any of the environments that
+might have been set up. For example, turning off VMs or deleting
+containers.
+
+The result of cleanup_exec does not affect job statuses. For example, a
+job will be marked as successful even if the following occurs:
+
+  Both prepare_exec and run_exec are successful.
+  cleanup_exec fails.
+
+The user can set cleanup_exec_timeout if they want to set some kind of
```

Comment:
ooooh - we should replace the current `CUSTOM_ENV_PRESERVE_*` behavior with a long (1 hour?) timeout instead of just skipping

Reply:
Mm yeah that would be good. An hour feels long to me, but I'm supposing it must not to you?

Reply:
Oh I misunderstood you when I first read this and thought you were just talking about adding a timeout to make sure broken jobs get cleaned up.

How are you thinking this would work? We could put something like a long sleep in the run step? It does seem like an easy way to take care of at least half of https://github.com/GSA-TTS/gitlab-runner-cloudgov/issues/34.

Reply:
Ideally it'd be something like what circleCI does, which is 10 minutes + whatever time you're ssh'd in to the runner to do your debugging. 1 hour was assuming that we wouldn't be able to have a dynamic timeout like that.

Reply:
> How are you thinking this would work?

🤷🏻 this might also be a misinterpretation of the `cleanup_exec_timeout` docs that triggered this whole thread. I was thinking that we could set that config and gitlab would take care of not calling cleanup right away, but that's probably wrong.

In any case, it was a thought for 34 or similar, not this PR

Reply:
I think I misunderstood it… I was thinking it was a timeout on how long it would wait to kill `run`, but it's a timeout on how long it will wait to kill `cleanup`. Every stage but `run` has one of the timeouts, and `run`'s time is managed in GitLab's settings globally and per runner.

But I do think we could use it to do the debugging `PRESERVE`'s better:
- set a long timeout on cleanup, just to override a shorter default if there is one
- set a sleep that's a bit shorter than the timeout
- do the regular cleanup after the sleep